src/Stack/Build/ConstructPlan.hs

```diff
@@ -426,9 +426,15 @@ addDep treatAsDep' name = do
                                 askPkgLoc = liftRIO $ do
                                   mrev <- getLatestHackageRevision name version
                                   case mrev of
-                                    Nothing -> error $ "No package revision found for: " <> show name
+                                    Nothing -> do
+                                      -- this could happen for GHC boot libraries missing from Hackage
```

Comment:
I found a comment in another part of the code which cleared up my misunderstanding the other day.

https://github.com/commercialhaskell/stack/blob/12d11a64f70cc41ec3709486df4ce5073e1620be/src/Stack/Build/Target.hs#L315-L318

So if stack _doesn't_ find a revision, that's not a problem. If it _does_ find a revision, it will use that. That's correct, then?

Reply:
Now when you ask it that way I'm not completely sure that it's 100% correct logic - the reason is that this `PIOnlyInstalled` should appear only for global libraries (GHC boot packages) and for those we don't have guarantees about proper versions/revisions on Hackage. From the other side as far as I understand we don't have any executables in those packages so practically `tellExecutablesUpstream` should be NOOP.